File: week4_story/StaircaseArchitect.cpp
/**
 * @filename   : StaircaseArchitect.cpp
 * @description: Calculate number of ways to climb stairs using recursion.
 * @author     : Ehtesham Nawaz
 */

#include <iostream>
using namespace std;

int countWays(int n)
{
    // Base cases
    if (n == 0 || n == 1)
        return 1;

    int jumpOneStair = countWays(n - 1);
    int jumpTwoStair = countWays(n - 2);

    return jumpOneStair + jumpTwoStair;
}

int main()
{
    int NthStair = 3;
    cout << countWays(NthStair) << endl;
    return 0;
}

File: Story4/StaircaseArchitect.java
/**

 * Description- A builder is designing a staircase with $N$ steps. He can climb either 1
 step or 2 steps at a time. Write a function to calculate the total number
 of distinct ways he can reach the top.
 * Author - Mayank Rathore

 */
public class StaircaseArchitect {
    static int countWays(int stepcount)
    {
        if(stepcount==1||stepcount==0)
        {
            return 1;

        }
        return countWays(stepcount - 1) + countWays(stepcount - 2);

    }
    public static void main(String[] args)
    {
        Input input=new Input();
        System.out.println("Enter number of stair");
        int stepcount= input.inputInteger();
        System.out.println("Nmber of ways are:"+countWays(stepcount));
    }

}
